kubernetes/updater/pkg/img/errors.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package img
+
+import (
+	"fmt"
+)
+
+// ImageNotValidatedError indicates that the validator did not validate the image.
+// It wraps the original error and implements trace.ErrorWrapper
+type ImageNotValidatedError struct {
+	err           error
+	validatorName string
+}
+
+// Error returns log friendly description of an error
+func (e *ImageNotValidatedError) Error() string {
+	return fmt.Sprintf("validator '%s' rejected the image: %s", e.validatorName, e.err.Error())
+}
+
+// OrigError return the original error
+func (e *ImageNotValidatedError) OrigError() error {
```

Comment:
Maybe use the `Unwrap` interface?

Reply:
OrigError implements the `trace.ErrorWrapper` interface, which allows the error to be unwrapped by `trace.Unwrap()`. I'm really not familiar with go's error management, is there another unwrapping go standard I should implement?

Reply:
https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/errors/wrap.go;l=16